tests/test_data.py

```diff
@@ -206,6 +207,26 @@ def test_indexing_into_sparse(self):
         sf.data[1:-1, 0] = np.arange(8)
         assert np.all(sf.data[1:-1, 0] == np.arange(8))
 
+    def test_uma_allocation(self):
```

Comment:
There's a `test_external_allocator` somewhere in this file. Could you move that test, and this test, within a new class, say  `TestAllocators`?

Reply:
I updated the PR with this change